apache/zeppelin
# [ZEPPELIN-2956] [font-end] Downloaded CSV/TSV data will get unexpected division when the column value contains both delimiter and quotation mark.

### What is this PR for?
Fix for ZEPPELIN-2956. This PR is to deal with the case when column value contains both delimiter and quotation mark. So the downloaded data can be well recogized and seperated by Excel.

### What type of PR is it?
Bug Fix

### Todos
None

### What is the Jira issue?
https://issues.apache.org/jira/browse/ZEPPELIN-2956

### How should this be tested?
Manual test.
After applying this PR, you can test the patch as following steps:
1、suppose one of the column value contain both delimiter and quotation mark:  
=>  hello  beijing  i  lo"v,"e  you

2、download the data as csv/tsv file
take download as csv format file =>  hello,beijing,"lo""v,""e",you

3、open the data file through Excel to verify the bug fix.

### Screenshots (if appropriate)
before patch:
<img width="335" alt="csvdata" src="https://user-images.githubusercontent.com/8579142/30922212-16f647ce-a36e-11e7-8217-8e697a86ca16.png">

after patch:
<img width="319" alt="afterpr" src="https://user-images.githubusercontent.com/8579142/30922240-2d1b65de-a36e-11e7-9d68-58cf3e621a8d.png">

### Questions:
* Does the licenses files need update?
no
* Is there breaking changes for older versions?
no
* Does this needs documentation?
no
